torch/distributed/fsdp/sharded_grad_scaler.py

```diff
@@ -118,45 +98,29 @@ def scale(self, outputs: Iterable[torch.Tensor]) -> Iterable[torch.Tensor]:
     def scale(
         self, outputs: Union[torch.Tensor, Iterable[torch.Tensor]]
     ) -> Union[torch.Tensor, Iterable[torch.Tensor]]:
-        if not self._enabled:
-            return outputs
-
-        if isinstance(outputs, torch.Tensor):
-            assert _is_supported_device(outputs)
-            if self._scale is None:
-                self._lazy_init_scale_growth_tracker(outputs.device)
-            assert self._scale is not None
-            scaled_output = outputs * self._scale.to(
-                device=outputs.device, non_blocking=True
+        scaled_outputs = super().scale(outputs)
+        # Here we ensure the return dtype is the same as the outputs dtype.
+        # For the FSDP + Mixed Precision use case, the loss output is in the Mixed Precision
+        # format (fp16, bf16) and so the scaled loss should be of the same dtype.
+        if isinstance(scaled_outputs, torch.Tensor):
+            return scaled_outputs.type(outputs.dtype)  # type: ignore[union-attr]
```

Comment:
Conceptual: For FSDP mixed precision, what is the dtype of `scaled_outputs` before calling `.type(outputs.dtype)`? What determines that dtype?